core/Util/plJPEG.cpp

```diff
@@ -195,7 +266,84 @@ void plJPEG::DecompressJPEG(hsStream* S, void* buf, size_t size)
     jpeg_finish_decompress(&ji.dinfo);
 }
 
-void plJPEG::CompressJPEG(hsStream* S, void* buf, size_t size)
+
+plMipmap* plJPEG::DecompressJPEG(hsStream* S)
+{
+    plJPEG& ji = Instance();
+
+    jpeg_hsStream_src(&ji.dinfo, S);
+    jpeg_read_header(&ji.dinfo, TRUE);
+    ji.dinfo.out_color_space = JCS_EXT_BGRA;    // Data stored as RGB on disk but Plasma uses BGR
+    jpeg_start_decompress(&ji.dinfo);
+
+    int row_stride = ji.dinfo.output_width * ji.dinfo.out_color_components;
+    int out_stride = row_stride;
+    RAII_JSAMPROW<1> jbuffer(row_stride);
+
+    // Start with a reasonable size for the buffer
+    std::vector<uint8_t> buffer(ji.dinfo.output_width * ji.dinfo.output_height * ji.dinfo.out_color_components);
+
+    size_t offs = 0;
+    while (ji.dinfo.output_scanline < ji.dinfo.output_height) {
+        while (offs + out_stride > buffer.capacity()) {
+            buffer.reserve(buffer.capacity() + INPUT_BUF_SIZE);
+        }
```

Comment:
```suggestion
```

As above, unless I'm misunderstanding something, we should already know the final uncompressed size and have that allocated.